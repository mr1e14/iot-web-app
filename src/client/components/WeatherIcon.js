import React from "react";
import matches from "../mediaQuery";
import clearDay from "../img/clear-day.png";
import clearNight from "../img/clear-night.png";
import cloudy from "../img/cloudy.png";
import defaultIcon from "../img/default.png";
import fog from "../img/fog.png";
import partlyCloudDay from "../img/partly-cloudy-day.png";
import partlyCloudyNight from "../img/partly-cloudy-night.png";
import rain from "../img/rain.png";
import sleet from "../img/sleet.png";
import snow from "../img/snow.png";
import thunderstorm from "../img/thunderstorm.png";
import tornado from "../img/tornado.png";
import wind from "../img/wind.png";

const icons = {
  "clear-day": clearDay,
  "clear-night": clearNight,
  cloudy,
  defaultIcon,
  fog,
  "partly-cloudy-day": partlyCloudDay,
  "partly-cloudy-night": partlyCloudyNight,
  rain,
  sleet,
  snow,
  thunderstorm,
  tornado,
  wind
};

const WeatherIcon = props => {
  const icon = props.weatherData ? props.weatherData.icon : "defaultIcon";
  const iconSize = matches("xs") ? "48px" : "64px";
  return (
    <img
      style={{ marginRight: "0.5rem" }}
      src={icons[icon]}
      height={iconSize}
      alt="weather-icon"
      align="middle"
    />
  );
};

export default WeatherIcon;
