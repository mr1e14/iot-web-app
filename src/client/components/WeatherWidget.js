import React from "react";
import Typography from "@material-ui/core/Typography";
import { WeatherIcon } from "./";

const WeatherWidget = props => {
  const { weatherData } = props;
  const temperature = weatherData ? parseInt(weatherData.temperature) : "";
  const humidity = weatherData ? parseInt(weatherData.humidity * 100) : "";
  return (
    <React.Fragment>
      <WeatherIcon {...props} />
      <Typography color="secondary">
        <span>
          {temperature}Â° | {humidity}%
        </span>
      </Typography>
    </React.Fragment>
  );
};

export default WeatherWidget;
