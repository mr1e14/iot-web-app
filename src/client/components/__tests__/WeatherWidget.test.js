import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

jest.mock("../mediaQuery");

import WeatherWidget from "../WeatherWidget";
import WeatherIcon from "../WeatherWidget";
import matches from "../mediaQuery";

const exampleWeatherData = {
  icon: "partly-cloudy",
  temperature: "15",
  humidity: "0.75"
};

describe("WeatherWidget", () => {
  let tree;
  let component;
  describe("when weatherData is supplied", () => {
    beforeEach(() => {
      component = renderer.create(
        <WeatherWidget weatherData={exampleWeatherData} />
      );
      tree = component.toJSON();
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe("when weatherData is not available", () => {
    beforeEach(() => {
      component = renderer.create(<WeatherWidget weatherData={null} />);
      tree = component.toJSON();
    });
    it("renders correctly", () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe("when renders", () => {
    let wrapper;
    describe("on xs devices", () => {
      beforeEach(() => {
        matches.mockReturnValue(false);
        wrapper = mount(<WeatherWidget weatherData={exampleWeatherData} />);
      });
      it("has one WeatherIcon", () => {
        expect(wrapper.find(WeatherIcon).length).toEqual(1);
      });
      it("presents weather data excluding humidity", () => {
        expect(
          wrapper
            .find("span")
            .at(1)
            .text()
        ).toBe("15°C");
      });
    });
    describe("on larger devices", () => {
      beforeEach(() => {
        matches.mockReturnValue(true);
        wrapper = mount(<WeatherWidget weatherData={exampleWeatherData} />);
      });
      it("has one WeatherIcon", () => {
        expect(wrapper.find(WeatherIcon).length).toEqual(1);
      });
      it("presents weather data including humidity", () => {
        expect(
          wrapper
            .find("span")
            .at(1)
            .text()
        ).toBe("15°C / 75%");
      });
    });
  });
});
