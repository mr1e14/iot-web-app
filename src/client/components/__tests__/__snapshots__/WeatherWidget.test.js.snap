// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherWidget when weatherData is not available renders correctly 1`] = `
<span
  className="MuiTypography-root-1 MuiTypography-caption-11"
>
  <img
    align="middle"
    alt="weather-icon"
    height="48px"
    src="test-file-stub"
    style={
      Object {
        "marginRight": "0.5rem",
      }
    }
  />
  <span>
    
    °C / 
    
    %
  </span>
</span>
`;

exports[`WeatherWidget when weatherData is supplied renders correctly 1`] = `
<span
  className="MuiTypography-root-1 MuiTypography-caption-11"
>
  <img
    align="middle"
    alt="weather-icon"
    height="48px"
    style={
      Object {
        "marginRight": "0.5rem",
      }
    }
  />
  <span>
    15
    °C / 
    75
    %
  </span>
</span>
`;
